Orange/widgets/data/owfile.py

```diff
@@ -194,11 +224,7 @@ def browse_file(self, in_demos=False):
             self, 'Open Orange Data File', start_file, self.dlg_formats)
         if not filename:
             return
-
-        self.add_path(filename)
-        self.source = self.LOCAL_FILE
-        self.load_data()
-
+        self.multiple_excelsheets(filename)
```

Comment:
This program flow is not so nice. The function is called `multiple_excelsheets` but it really handles all types of files? I'd comparatively prefer an `is_excel(filename)` method and the rest of the lines remaining here.